src/lib/communication/DY.Lib.Communication.Core.Invariants.fst

```diff
@@ -128,34 +129,29 @@ let default_comm_higher_layer_event_preds (a:Type) {| parseable_serializeable by
 }
 
 #push-options "--ifuel 1 --fuel 0"
-let event_predicate_communication_layer 
+let event_predicate_communication_layer
   {|cinvs:crypto_invariants|}
   (#a:Type) {| parseable_serializeable bytes a |}
-  (higher_layer_preds:comm_higher_layer_event_preds a) : 
+  (higher_layer_preds:comm_higher_layer_event_preds a) :
   event_predicate communication_event =
   fun tr prin e ->
     (match e with
     | CommConfSendMsg sender receiver payload -> (
       is_knowable_by (comm_label sender receiver) tr payload /\
-      (match parse a payload with
-      | None -> False
-      | Some payload_p -> higher_layer_preds.send_conf tr sender receiver payload_p
-      )
+      parse_and_pred (higher_layer_preds.send_conf tr sender receiver) payload
     )
     | CommConfReceiveMsg receiver payload -> (
-      exists sender.
-        (
-          event_triggered tr sender (CommConfSendMsg sender receiver payload) \/
-          is_publishable tr payload
-        )
+      Some? (parse a payload) /\
```

Comment:
Is this line necessary? I don't see where the message gets parsed here.

Reply:
This line (along with `parse_wf_lemma`) makes it so that the receive event contains enough information to prove properties about the fields of the message, even in the attacker case. It is used, for instance, in line 65 of DY.Lib.Communication.Core.Properties.fst.